mlserver/parallel/pool.py

```diff
@@ -92,6 +97,10 @@ async def _process_responses(self):
     async def _process_response(self, response: InferenceResponseMessage):
         internal_id = response.id
 
+        queue_size = self._requests.qsize()
```

Comment:
This one has changed a bit on the `master` branch. There's no longer a single `_requests` queue - instead, there's a separate queue per worker. This could be just added as a label on the metric though (i.e. instead of diff metric names for each worker).

Reply:
Perfect thank you. Just one question, how can I add each worker queue in only one label, just with one metric in the class Worker?

Reply:
If I understood that right, I guess the options would be:

1. Create the metric in the `Worker.__init__()` method. Main question is whether the `prometheus_client` library is OK with having the same metric name created multiple times. If that works, we could just track each request on the `Worker.send_request` method, and the label could be something like `worker=self.pid`. 
2. Create a single metric in the (new) `Dispatcher.__init__()` class, and track each request on the `Dispatcher.dispatch` method. The label could also be set like `worker=worker.pid`. 

I think `2.` is more similar to what you currently have in this PR, so it's probably an easier option. 

PS: While writing this comment, I also realised that TBH it's not that big of a deal that we label the metrics with the worker index / pid. So we could even go with `2.` and don't label the metrics.

Reply:
We ended up choosing option 1

Reply:
Thank you for the guide!